config/initializers/active_record.rb

```diff
@@ -40,36 +40,28 @@ def to_json(options = {})
 
     def self.conditions_for_date(column, date)
       if date == 'today'
-        return ["#{column}::DATE = ?", Date.today.to_s]
+        return ["#{column} >= ? AND #{column} < ?", Date.today.to_s, Date.tomorrow.to_s]
       end
       year, month, day = date.to_s.split('-').map do |d|
         d = d.blank? ? nil : d.to_i
         d == 0 ? nil : d
       end
-      if date.to_s =~ /^\d{4}/ && year && month && day
-        datestring = "#{year}-#{month}-#{day}"
+      if year || month || day
         begin
-          Date.parse(datestring)
-          ["#{column}::DATE = ?", datestring]
+          if day
+            date = Date.parse("#{ year }-#{ month }-#{ day }")
+            [ "#{column} >= ? AND #{column} < ?", date, date + 1.day ]
+          elsif month
+            date = Date.parse("#{ year }-#{ month }-1")
+            [ "#{column} >= ? AND #{column} < ?", date, date + 1.month ]
```

Comment:
No. How was that working before? Looks like it takes a date string and splits it into year-month-day. What do you send to get it to search on month? Something like "-3-" to search for March? Can they do the same on day (e.g. "--5" to get things records on the 5th of the month)? Presumably we are using and need to keep that functionality? Can you think of a place we take advantage of this feature that I can test with?